crates/storage/provider/src/providers/static_file/writer.rs

```diff
@@ -547,6 +547,44 @@ impl StaticFileProviderRW {
         Ok(result)
     }
 
+    /// Appends multiple receipts to the static file.
+    ///
+    /// Returns the current [`TxNumber`] as seen in the static file.
+    pub fn append_receipts<I>(&mut self, receipts: I) -> ProviderResult<TxNumber>
+    where
+        I: IntoIterator<Item = Result<(TxNumber, Receipt), ProviderError>>,
+    {
+        let mut receipts_iter = receipts.into_iter().peekable();
+        // If receipts are empty, we can simply return the current TxNumber as seen in the static
+        // file.
+        if receipts_iter.peek().is_none() {
+            return Ok(self.writer.user_header().tx_end().expect("qed"));
+        }
+
+        let start = Instant::now();
+        self.ensure_no_queued_prune()?;
+
+        // At this point receipts contains at least one receipt, so this would be overwritten.
+        let mut tx_number = 0;
+        let mut count = 0;
+
+        for receipt_result in receipts_iter {
+            let (tx_num, receipt) = receipt_result?;
+            tx_number = self.append_with_tx_number(StaticFileSegment::Receipts, tx_num, receipt)?;
+            count += 1;
+        }
+
+        if let Some(metrics) = &self.metrics {
+            metrics.record_segment_operation(
+                StaticFileSegment::Receipts,
+                StaticFileProviderOperation::AppendAverage,
+                Some(start.elapsed() / count as u32),
+            );
```

Comment:
we don't want to have a separate operation type `append-average`, because it's the same as `append`, but just for batched writes. What we want is to be able to increment the counter metric by `count` and record the `start.elapsed() / count` in the histogram metric.

I propose to add a new `metrics.record_segment_operations` that will do that by accepting additional `count` argument. In the implementation, it will divide the `duration` by `count`, reporting the average to the histogram

Reply:
@kamuik16 I believe it's not resolved yet

Reply:
Ohh, so sorry, I guess I forgot to push the changes.

Reply:
> @kamuik16 I believe it's not resolved yet

```rust
        if let Some(metrics) = &self.metrics {
            metrics.record_segment_operations(
                StaticFileSegment::Receipts,
                StaticFileProviderOperation::AppendAverage,
                count,
                Some(start.elapsed()),
            );
        }
```
```rust
pub(crate) fn record_segment_operations(
```

Doesn't this solve this?

Reply:
we don't need a separate `AppendAverage` for that, can use just the `Append` because we report the average

Reply:
> we don't need a separate `AppendAverage` for that, can use just the `Append` because we report the average calculated by us

so just replace AppendAverage with Append, right? rest seem fine?

Reply:
fixed in https://github.com/paradigmxyz/reth/pull/8718/commits/f49dbe9ca85e6945337d4272e8f80c7ceafd5aab